presto-main/src/main/java/com/facebook/presto/execution/scheduler/SourcePartitionedScheduler.java

```diff
@@ -429,6 +420,15 @@ public void close()
         return result.build();
     }
 
+    @Override
+    public void notifyAllLifespansFinishedExecution()
+    {
+        checkState(groupedExecution);
+        state = State.FINISHED;
+        splitSource.close();
+        whenFinishedOrNewLifespanAdded.set(null);
```

Comment:
What does "finished" mean in `whenFinishedOrNewLifespanAdded`? It seems to mean scheduling finished for ungrouped execution and execution finished for grouped execution?

Reply:
From the code, it previously mean "finishing schedule". Now it also get triggered when all lifespan execution finished ...

To be more clear, previously it get triggered:
- When a single lifespan finish schedule splits
- When all lifespan finish schedule (i.e. noMoreLIfespans get called)

No it means
- When a single lifespan finish schedule splits
- When all lifespan finish schedule (i.e. all lifespans finished execution)